src/OnboardingSPA/pages/Steps/GetStarted/SiteTypeSetup/SecondarySite/index.js

```diff
@@ -29,154 +29,219 @@ const StepPrimarySetup = () => {
 		setIsDrawerSuppressed( true );
 		setDrawerActiveView( VIEW_NAV_GET_STARTED );
 		setIsHeaderNavigationEnabled( true );
+		getSiteClassificationsData();
 	}, [] );
 
-	const [ clickedIndex, changeCategory ] = useState( -1 );
+	const contents = getContents();
+	const defaultPrimaryType = 'business';
+	const [ siteClassData, setSiteClassData ] = useState();
+	const [ primaryTypeList, setPrimaryTypeList ] = useState();
+	const [ primaryType, changePrimaryType ] = useState( defaultPrimaryType );
+	const [ secondaryType, changeSecondaryType ] = useState( '' );
 	const [ inputCategVal, changeInputCateg ] = useState( '' );
 
-	const { currentStep, currentData } = useSelect( ( select ) => {
+	const { currentData } = useSelect( ( select ) => {
 		return {
-			currentStep: select( nfdOnboardingStore ).getCurrentStep(),
 			currentData:
 				select( nfdOnboardingStore ).getCurrentOnboardingData(),
 		};
 	}, [] );
 
-	const selectedCategoryInStore = currentData?.data?.siteType?.secondary;
-	const categoriesArray = content.categories;
-	const subCategories = categoriesArray[ 0 ]?.subCategories;
+	/**
+	 * Function which fetches the Site Classifications
+	 *
+	 */
+	const getSiteClassificationsData = async () => {
+		let siteClassificationsData = await getSiteClassifications();
+		if ( siteClassificationsData?.body.length === 0 ) {
+			siteClassificationsData = fallbackDataTypes;
+		}
+		setSiteClassData( siteClassificationsData?.body );
+		setPrimaryTypeList(
+			Object.keys( siteClassificationsData?.body?.types )
+		);
 
-	/**This condition fills the data in input box if the saved category isn't a subcategory from the content*/
-	if (
-		selectedCategoryInStore &&
-		! inputCategVal &&
-		subCategories.indexOf( selectedCategoryInStore ) === -1
-	) {
-		if ( selectedCategoryInStore !== 'secondaryCategory' )
-			changeInputCateg( selectedCategoryInStore );
-	}
+		if ( currentData?.data?.siteType?.primary !== '' )
+			changePrimaryType( currentData?.data?.siteType?.primary );
+		changeSecondaryType( currentData?.data?.siteType?.secondary ?? '' );
+
+		if ( currentData?.data?.siteType?.labelPri === 'custom' ) {
+			changePrimaryType( defaultPrimaryType );
+			changeSecondaryType( '' );
+			changeInputCateg( currentData?.data?.siteType?.secondary );
+		}
+
+		if ( currentData?.data?.siteType?.labelSec === 'custom' ) {
+			changePrimaryType( defaultPrimaryType );
+			changeSecondaryType( '' );
+			changeInputCateg( currentData?.data?.siteType?.secondary );
+		}
```

Comment:
We can easily simplify this logic, ref my comments on the new data structure and moving inputs to a single state.